keps/sig-api-machinery/storage-migration-auto-trigger.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Automated Storage Version Migration with Storage Version Hash
+authors:
+  - "@xuchao"
+owning-sig: sig-api-machinery
+reviewers:
+  - "@deads2k"
+  - "@yliaog"
+approvers:
+  - "@deads2k"
+  - "@lavalamp"
+creation-date: 2019-01-23
+last-updated: 2019-01-23
+status: provisional
+---
+
+## Table of Contents
+
+* [Goal](#goal)
+* [API design](#api-design)
+* [Storage migration triggering controller](#storage-migration-triggering-controller)
+* [Implications to cluster operators](#implications-to-cluster-operators)
+* [Future work: HA clusters](#future-work-ha-clusters)
+
+## Goal
+
+As the discovery document now exposes the [storage version hash][], storage version
+migration should start automatically when the hash changes.
+
+[storage version hash]:https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/35-storage-version-hash.md
+
+## API design
+
+We introduce the `StorageState` API, as a CRD defined in the `migration.k8s.io`
```

Comment:
Did we already use this group name for anything? I'd prefer to make it more specific if not. There are many possible sorts of migrations we might do.

Reply:
This is the group name we picked for [StorageVersionMigration](https://github.com/kubernetes-sigs/kube-storage-version-migrator/blob/master/pkg/apis/migration/v1alpha1/types.go#L29), which we used to make migration request and tracking migration progress.

Reply:
How about changing the API name to `StorageVersionState` to be more specific? It's also symmetric to `StorageVersionMigration`.

Reply:
That sounds like good consistency.